harness-delegate-ng/templates/ccm/cost-access.yaml

```diff
@@ -0,0 +1,76 @@
+{{- if .Values.ccm.visibility }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ .Values.ccm.role_name }}
```

Comment:
Your role and role bindings needs to be per delegate and include delegate name. 
You are making it a shared resource between all delegates, so this way if you use helm to delete one delegate it would also remove the cluster role and the binding and break other delegates. Please check how other roles and bindings are implemented and test delete and upgrade scenarios.